packages/react-dom/src/__tests__/ReactDOMHydrationDiff-test.js

```diff
@@ -103,6 +103,12 @@ describe('ReactDOMServerHydration', () => {
         It can also happen if the client has a browser extension installed which messes with the HTML before React loaded.
 
         https://react.dev/link/hydration-mismatch
+
+          <Mismatch isClient={true}>
+            <div className="parent">
+              <main className="child">
+        +       client
+        -       server
```

Comment:
Is there a reason `+` added lines are printed before `-` removed lines? Usually in diff visualizations, removed lines are printed first.

Reply:
@sompylasar nice to bump into you here haha, it's been a while since you were adding this...

Reply:
There’s an awkward scenario that happens at the end where if we have more client siblings we don’t necessarily know if they would’ve matched so we cut off next siblings with “…”. However if there are extra server nodes at the tail then we do list them as removed next siblings which puts them at the end. Sometimes a swap looks like this scenario so I just followed that ordering to avoid sometimes flipping it.

I don’t think it’s necessarily inherent though. Could probably be changed.